lib/location/method_cellular.c

```diff
@@ -0,0 +1,207 @@
+			"Connected - home network" : "Connected - roaming");
+		lte_connected = true;
+		break;
+	case LTE_LC_EVT_NEIGHBOR_CELL_MEAS: {
+		struct lte_lc_cells_info cells = evt->cells_info;
+		struct lte_lc_cell cur_cell = cells.current_cell;
+
+		LOG_INF("Cell measurements results received");
+		memset(&cell_data, 0, sizeof(struct lte_lc_cells_info));
+
+		if (cur_cell.id) {
+			/* Copy current and neighbor cell information */
+			memcpy(&cell_data, &evt->cells_info, sizeof(struct lte_lc_cells_info));
+			memcpy(neighbor_cells, evt->cells_info.neighbor_cells,
+				sizeof(struct lte_lc_ncell) * cell_data.ncells_count);
+			cell_data.neighbor_cells = neighbor_cells;
+		} else {
+			LOG_INF("No current cell information from modem.");
+		}
+		if (!cells.ncells_count) {
+			LOG_INF("No neighbor cell information from modem.");
+		}
+		k_sem_give(&cellmeas_data_ready);
+	} break;
+	default:
+		break;
+	}
+}
+
+static int method_cellular_ncellmeas_start(void)
+{
+	int err;
+
+	LOG_INF("Triggering start of cell measurements");
+	err = lte_lc_neighbor_cell_measurement();
+	if (err) {
+		LOG_ERR("Failed to initiate neighbor cell measurements: %d", err);
+		loc_core_event_cb_error();
+		running = false;
+		return err;
+	}
+	return 0;
+}
+
+static void method_cellular_positioning_work_fn(struct k_work *work)
+{
+	struct multicell_location location;
+	struct loc_location location_result = { 0 };
+	int ret;
+
+	ARG_UNUSED(work);
+
+	ret = method_cellular_ncellmeas_start();
+	if (ret) {
+		LOG_WRN("Cannot start cell measurements - cannot fetch location");
+		loc_core_event_cb_error();
+		running = false;
+		return;
+	}
+
+	ret = k_sem_take(&cellmeas_data_ready, K_SECONDS(60)); /* TODO: timeout to params? */
+
+	if (ret) {
+		LOG_WRN("Timeout for waiting cell measurements - cannot fetch location");
+		loc_core_event_cb_error();
+		running = false;
+	} else if (running) {
+		if (cell_data.current_cell.id == 0) {
+			LOG_WRN("No cells were found - cannot fetch location");
+			loc_core_event_cb_error();
+			running = false;
+			return;
+		}
+
+		ret = multicell_location_get(&cell_data, &location);
+		if (ret) {
+			LOG_ERR("Failed to acquire location from multicell_location lib, error: %d",
+				ret);
+			loc_core_event_cb_error();
+			running = false;
+		} else {
+			location_result.latitude = location.latitude;
+			location_result.longitude = location.longitude;
+			location_result.accuracy = location.accuracy;
+			(void)lte_lc_neighbor_cell_measurement_cancel();
+			if (running) {
+				loc_core_event_cb(&location_result);
+				running = false;
+			}
+		}
+	}
+}
+
+int method_cellular_location_get(const struct loc_method_config *config)
+{
+	const struct loc_cellular_config *cellular_config = &config->config.cellular;
+
+	ARG_UNUSED(cellular_config);
+
+	LOG_INF("Starting to get a location by using cellular method");
+
+	if (running) {
+		LOG_ERR("Previous operation on going.");
+		return -EBUSY;
+	}
+
+	k_work_submit(&method_cellular_positioning_work);
```

Comment:
There's a dedicated work queue for location API which we could utilize throughout the library:
`k_work_submit_to_queue(loc_core_work_queue_get(), &method_cellular_positioning_work);`
@jhirsi

Reply:
This has been done